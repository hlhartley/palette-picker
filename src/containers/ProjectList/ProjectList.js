import React, { Component } from 'react';
import { connect } from 'react-redux';
import ProjectCard from '../../components/ProjectCard/ProjectCard';
import uuid from 'uuid/v4';

export class ProjectList extends Component {
  render() {
    return (
      <div className="ProjectList">
        <p className="projectlist-subtitle">Your Creations:</p>
        {
          this.props.projects.map(project => {
            return <ProjectCard key={uuid()} />
          })
        }
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  projects: state.projects
});

export default connect(mapStateToProps)(ProjectList);
