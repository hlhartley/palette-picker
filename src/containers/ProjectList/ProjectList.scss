.ProjectList {
    border: 2px solid tomato;
    display: flex;
    flex-direction: column;
}