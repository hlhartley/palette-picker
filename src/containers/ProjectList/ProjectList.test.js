import React from 'react';
import { shallow } from 'enzyme';
import { ProjectList } from './ProjectList';

const projectsMock = { id: 1, name: 'testProject' };
const palettesMock = { id: 1, project_id: 1, name: 'testPalette' };

describe('ProjectList', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
        <ProjectList projects={projectsMock} palettes={palettesMock} />
      )
    });
  it('should properly render the component elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});