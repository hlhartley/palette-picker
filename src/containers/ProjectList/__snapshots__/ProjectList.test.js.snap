// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList should properly render the component elements 1`] = `undefined`;
